smithy-trait-codegen/src/main/java/software/amazon/smithy/traitcodegen/TraitCodegenSettings.java

```diff
@@ -55,8 +55,8 @@ public final class TraitCodegenSettings {
     ) {
         this.packageName = Objects.requireNonNull(packageName);
         this.smithyNamespace = Objects.requireNonNull(smithyNamespace);
-        if (smithyNamespace.startsWith(SMITHY_API_NAMESPACE)) {
-            throw new IllegalArgumentException("The `smithy` namespace is reserved.");
+        if (smithyNamespace.equals(SMITHY_API_NAMESPACE) || smithyNamespace.startsWith(SMITHY_API_NAMESPACE + ".")) {
```

Comment:
actually, `SMITHY_API_NAMESPACE` is set to `smithy`. Shouldn't it just be `smithy.api`?